modules/ocs2/src/main/scala/gem/ocs2/DoobieClient.scala

```diff
@@ -0,0 +1,40 @@
+package gem.ocs2
+
+import doobie.contrib.postgresql.syntax._
+import doobie.imports._
+
+import java.util.logging.{Level, Logger}
+
+import scalaz._
+import Scalaz._
+import scalaz.concurrent.Task
+import scalaz.effect.IO
+
+/** Shared support for import applications using Doobie. */
+trait DoobieClient {
+
+  val xa = DriverManagerTransactor[IO](
+    "org.postgresql.Driver",
+    "jdbc:postgresql:gem",
+    "postgres",
+    ""
+  )
+
+  val lxa = DriverManagerTransactor[Task](
+    "org.postgresql.Driver",
+    "jdbc:postgresql:gem",
+    "postgres",
+    ""
+  )
+
+  def configureLogging(): Unit = List(
+    "edu.gemini.spModel.type.SpTypeUtil"
+  ).map(Logger.getLogger).foreach(_.setLevel(Level.OFF))
+
+  def ignoreUniqueViolation(fa: ConnectionIO[Int]): ConnectionIO[Int] =
+    for {
+      s <- HC.setSavepoint
+      n <- fa.onUniqueViolation(HC.rollback(s).as(0))
+      _ <- HC.releaseSavepoint(s)
```

Comment:
N.B. this will leak the savepoint if there is a failure other than unique violation. This is probably fine because the whole transaction will be rolled back in that case, but the pedantic way to do this would be `… as(0)) ensuring HC.releaseSavepoint …`.